### PR TITLE
Frontend: UpdateBackfill and DeleteBackfill handlers

### PR DESCRIPTION
**Pull request by aLekSer**
**What this PR does / Why we need it**:

UD methods for Backfill structure.

**Which issue(s) this PR fixes**:
Work on #1240

**Special notes for your reviewer**:
This PR contains another PR with CRud part of Frontend: #1279 

So review please only changes from this commit and later commits:
https://github.com/googleforgames/open-match/pull/1292/commits/9f970a7a8b85ebe2018743c4344a671614c7c810

### CODE REVIEW

File: internal/app/frontend/frontend_service.go
Comment (hsorellana) [inline internal/app/frontend/frontend_service.go:333]: this comment is unrelated here

File: internal/app/frontend/frontend_service.go
Comment (yeukovichd) [inline internal/app/frontend/frontend_service.go:109]: Why is this check required? If we look at doCreateBackfill it always assigns new value to CreateTime.

File: internal/app/frontend/frontend_service.go
Comment (yeukovichd) [inline internal/app/frontend/frontend_service.go:165]: Maybe it is better to use WithError and specify fields such as backfill_id?

File: internal/app/frontend/frontend_service.go
Comment (aLekSer) [reply internal/app/frontend/frontend_service.go:109]: This is a leftover of CreateTicket and is part of #1279 This PR would be rebased on top of master after #1279 would be merged.

**Comment by aLekSer** [comment]
Unrelated to the change CI failure:
```

time="2020-11-26T00:14:25Z" level=error msg="context canceled" app=openmatch component=api.test
--- FAIL: TestMatchFunctionMatchCollision (0.43s)
    fetch_matches_test.go:158: 
        	Error Trace:	fetch_matches_test.go:158
        	Error:      	Should be true
        	Test:       	TestMatchFunctionMatchCollision
        	Messages:   	381.29587ms
```

File: internal/app/frontend/frontend_service.go
Comment (Laremere) [inline internal/app/frontend/frontend_service.go:174]: This must be done after the ticket is updated:
If the server performs this action, then dies, the tickets will think they can return to the active pool and find other matches, but the backfill will still think it can put tickets in this backfill, potentially assigning them twice.

The other way around, the backfill will release it's claim in the tickets, and if this server dies, then the tickets will just time out and then re-enter the pool.

File: internal/app/frontend/frontend_service.go
Comment (Laremere) [inline internal/app/frontend/frontend_service.go:149]: The update here should be done defensively, currently you perform the get, but use the backfill passed in as the source of truth.  Instead start with the backfill from the get, and then update each field which should be updated.  Eg, create time would be overwritten if the wrong value were sent here.

File: internal/testing/e2e/backfill_test.go
Comment (Laremere) [inline internal/testing/e2e/backfill_test.go:86]: To better test this method, send an entirely new Backfill protobuf object, with the new value but setting the ID to the one from createdBf.

As it is, this test incorrectly thinks it verifies CreateTime can't be updated.

Comment (Laremere) [reply internal/testing/e2e/backfill_test.go:86]: Also it would be a better way to show that generation doesn't need to be right.

File: internal/testing/e2e/backfill_test.go
Comment (Laremere) [inline internal/testing/e2e/backfill_test.go:82]: Test generation values?

**Review by Laremere — CHANGES_REQUESTED** [review]

File: internal/app/frontend/frontend_service.go
Comment (aLekSer) [reply internal/app/frontend/frontend_service.go:149]: Updated the code with:

```
	// Update generation here, because Frontend is used by GameServer only
	bfStored.Generation++
	bfStored.SearchFields = backfill.SearchFields
	bfStored.Extensions = backfill.Extensions
```

**Comment by aLekSer** [comment]
All comments are resolved.

File: internal/app/frontend/frontend_service.go
Comment (Laremere) [inline internal/app/frontend/frontend_service.go:125]: Why this comment?  It's entirely redundant with the line below?

**Review by Laremere — APPROVED** [review]
Extra comment should be removed, but a few things are depending on this so I'll merge it if the tests pass.

File: internal/app/frontend/frontend_service.go
Comment (aLekSer) [reply internal/app/frontend/frontend_service.go]: That's true updated

**Comment by Laremere** [comment]
Tests are still failing....

**Pull request merged**
